### ISSUE TITLE
Please provide some docs for GlobalConfigProvider

### ISSUE THREAD

**Issue by thevery** [robolectric/robolectric#4770]
http://robolectric.org/migrating/#migrating-to-40 doesn't have any unfortunately

**Comment by castorflex** [comment]
Hell @thevery , after some digging I found this https://github.com/robolectric/robolectric/blob/ecb76bc5240e242ec5e5c54cbe4c72683adfb034/pluginapi/src/main/java/org/robolectric/pluginapi/package-info.java

It seems this is the way to provide your own `GlobalConfigProvider`. Example:

```java
@AutoService(GlobalConfigProvider.class)
public class MyOwnConfigProvider implements GlobalConfigProvider {
    @Override
    public Config get() {
        return new Config.Builder()
                .setApplication(MyApplication.class)
                .build();
    }
}
```

You will also need to add AutoService:

```groovy
    testCompileOnly libraries.auto_service
    testAnnotationProcessor libraries.auto_service
```

Hope this will help you :)

**Comment by thevery** [comment]
@castorflex thanks!

Actually I've tried naive implementation and it doesn't work. I'll try again later.
Another question is dynamic configuration, i.e. having different `RobolectricTestRunner`s for different tests.

**Comment by xian** [comment]
@thevery — what are you trying to do that you need different `RobolectricTestRunner`s?

**Comment by thevery** [comment]
Moreover, runners are broken in 4.x, see https://github.com/robolectric/robolectric/issues/4779

**Comment by franzueto** [comment]
I'm not able to make a `GlobalConfigProvider` work. I'm using 4.2.1 version and I'm following @castorflex guidance:

```java
@AutoService(GlobalConfigProvider.class)
public class MyGlobalConfigProvider implements GlobalConfigProvider {

    @Override
    public Config get() {
        return new Config.Builder()
            .setApplication(TestApplication.class)
            .build();
    }
}
```

But if I try using the annotation directly, it works.
```java
@Config(application = TestApplication.class)
@RunWith(RobolectricTestRunner.class)
```
Am I missing something? Any help?

**Comment by SirWellington** [comment]
Yes this is confusing. Why was this made more complicated then before?

**Comment by frapontillo** [comment]
We do need some kind of documentation, as simply annotating my `GlobalConfigProvider` implementation doesn't seem to do anything at all.
